tests/LanguageServer/Example/TestExtension.php

```diff
@@ -70,18 +74,31 @@ public function test()
         }, [ LanguageServerExtension::TAG_SERVICE_PROVIDER => []]);
 
         $container->register('test.command', function (Container $container) {
-            return new class implements CoreCommand {
-                public function __invoke(string $text): Promise
-                {
-                    return new Success($text);
-                }
-            };
+            return new ClosureCommand(function (string $text): Promise {
+                return new Success($text);
+            });
         }, [
             LanguageServerExtension::TAG_COMMAND => [
                 'name' => 'echo',
             ],
         ]);
 
+        $container->register('test.progress_notifier_handler', function (Container $container) {
+
+            return new ClosureHandler('test/progress_notifier', function () use ($container){
+                $notifier = $container->get(ProgressNotifier::class);
+                assert($notifier instanceof ProgressNotifier);
+                $token = WorkDoneToken::generate();
+                $notifier->begin($token, 'Test');
```

Comment:
@camilledejoye this fails with the message notifier because $message is optional, fixing in language-serveer package